build.gradle

```diff
@@ -82,7 +83,7 @@ dependencies {
     // Add android plugin to the test classpath, so that we can jump into class definitions,
     // read their sources, set break points, etc while debugging android unit tests.
     // Change the version if necessary to match the specific unit test.
-    testRuntime 'com.android.tools.build:gradle:2.3.0'
+    testCompile 'com.android.tools.build:gradle:3.1.2'
```

Comment:
I think if we follow the pattern in the existing android tests we can construct a project with an android version on a per test project basis. Then we can validate that the new code does not cause problems for android plugin 2.x.

Reply:
Yes, validating backwards compat is a requirement for all versions we support. Maybe we some point we would drop support for 2.x but for now I think we should validate all new code for 2.x versions.

I think the tests would probably end up looking something like [BuildLogicFunctionalTest.java](https://docs.gradle.org/current/userguide/test_kit.html)

Reply:
I've done tests that way in some other projects I was just trying to be too lazy here. I'll go revamp these tests to use the external run api's which should get rid of the need for the version check `Require` on the 3.+ test as well.